official/projects/longformer/sentence_prediction_with_checkpoint_convert.py

```diff
@@ -0,0 +1,388 @@
+# Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Sentence prediction (classification) task."""
+import dataclasses
+from typing import List, Union, Optional
+
+from absl import logging
+import numpy as np
+import orbit
+from scipy import stats
+from sklearn import metrics as sklearn_metrics
+import tensorflow as tf
+
+from official.core import base_task
+from official.core import config_definitions as cfg
+from official.core import task_factory
+from official.modeling import tf_utils
+from official.modeling.hyperparams import base_config
+from official.nlp.configs import encoders
+from official.nlp.data import data_loader_factory
+from official.nlp.modeling import models
+from official.nlp.tasks import utils
+
+import pickle
+
+METRIC_TYPES = frozenset(
+    ['accuracy', 'matthews_corrcoef', 'pearson_spearman_corr'])
+
+
+@dataclasses.dataclass
+class ModelConfig(base_config.Config):
+  """A classifier/regressor configuration."""
+  num_classes: int = 0
+  use_encoder_pooler: bool = False
+  encoder: encoders.EncoderConfig = encoders.EncoderConfig()
+
+
+@dataclasses.dataclass
+class SentencePredictionConfig(cfg.TaskConfig):
+  """The model config."""
+  # At most one of `init_checkpoint` and `hub_module_url` can
+  # be specified.
+  init_checkpoint: str = ''
+  init_cls_pooler: bool = False
+  initial_parameters_from_pk: str = ''
+  hub_module_url: str = ''
+  metric_type: str = 'accuracy'
+  # Defines the concrete model config at instantiation time.
+  model: ModelConfig = ModelConfig()
+  train_data: cfg.DataConfig = cfg.DataConfig()
+  validation_data: cfg.DataConfig = cfg.DataConfig()
+
+
+@task_factory.register_task_cls(SentencePredictionConfig)
+class SentencePredictionTask(base_task.Task):
```

Comment:
Consider avoiding forking the task.
We can move initialize() to a standalone binary to save a tf checkpoint.
Or just add a write after initialization and keep this file for your local conversion use.